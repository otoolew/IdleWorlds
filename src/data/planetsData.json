{
	"planets": [
		{
			"id": "5b7aea183bf8367b5476c9e6",
			"name": "DOLORE",
			"isActive": true,
			"wealth": "$64,757,168,235.05",
			"age": 784890749
		},
		{
			"id": "5b7aea187ca55f9ce1c23845",
			"name": "ELIT",
			"isActive": true,
			"wealth": "$315,977,512,510.90",
			"age": 659095575
		},
		{
			"id": "5b7aea1884105003d46450a9",
			"name": "EXCEPTEUR",
			"isActive": true,
			"wealth": "$240,577,666,383.70",
			"age": 266767889
		},
		{
			"id": "5b7aea181b1ad23a7dd97378",
			"name": "EU",
			"isActive": true,
			"wealth": "$288,965,195,305.90",
			"age": 577478057
		},
		{
			"id": "5b7aea18c0ca6969d793bf1e",
			"name": "LABORIS",
			"isActive": true,
			"wealth": "$115,087,829,552.91",
			"age": 205891012
		},
		{
			"id": "5b7aea182007ede9f8ccb27e",
			"name": "OFFICIA",
			"isActive": false,
			"wealth": "$249,686,953,867.76",
			"age": 264638419
		},
		{
			"id": "5b7aea1853ec041bc301ede5",
			"name": "EST",
			"isActive": false,
			"wealth": "$344,884,597,724.36",
			"age": 89802163
		}
	]
}