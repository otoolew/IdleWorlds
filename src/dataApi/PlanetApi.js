class PlanetApi{
    constructor(rawData){
        this.rawData = rawData;
        console.log(this.rawData);
    }
    mapToObject(arr){
        return arr.reduce((acc, curr)=>{
            acc[curr.id] = curr;
            return acc;
        },{});
    }
    getPlanets(){
        return this.mapToObject(this.rawData.planets);
    }
}
export default PlanetApi;