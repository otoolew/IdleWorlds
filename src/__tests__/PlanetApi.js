import PlanetApi from '../dataApi/PlanetApi';
import {data} from '../data/planetsData.json';

const api = new PlanetApi(data);
describe('PlanetApi',()=>{
    it('exposes planets as an object', ()=>{
        const planets = api.getPlanets();
        const planetId = data.planets[0].id;
        const planetName = data.planets[0].planets.name;
        expect(planets).toHaveProperty(planetId);
        expect(planets.name).toBe(planetName);
    });
});