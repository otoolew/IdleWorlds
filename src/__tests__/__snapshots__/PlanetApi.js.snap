// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mouse changes the class when hovered 1`] = `<div />`;
